From 8f1f71f8cde88f1564da7babfeb5177f14773075 Mon Sep 17 00:00:00 2001
Message-ID: <8f1f71f8cde88f1564da7babfeb5177f14773075.1732384228.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1732384228.git.sharlatanus@gmail.com>
References: <cover.1732384228.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Thu, 21 Nov 2024 19:38:04 +0000
Subject: [PATCH 43/46] gnu: python-synphot: Update to 1.5.0.

* gnu/packages/astronomy.scm (python-synphot): Update to 1.5.0.
Simplify package.
[arguments]<phases>: Use default 'check phase.
[native-inputs]: Add python-setuptools and python-wheel.

Change-Id: Iccc0c2bedc7075af3da2bc0ffb822787d40d6f41
---
 gnu/packages/astronomy.scm | 25 +++++++++----------------
 1 file changed, 9 insertions(+), 16 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 1295b04f6b..aec9cc330a 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -5881,36 +5881,29 @@ (define-public python-suntime
 (define-public python-synphot
   (package
     (name "python-synphot")
-    (version "1.4.0")
+    (version "1.5.0")
     (source
      (origin
        (method url-fetch)
        (uri (pypi-uri "synphot" version))
        (sha256
-        (base32 "0fk860bxaqbamrq15pkqlbvhbf70y50cra2mgvv9r0cxq37isbi6"))))
+        (base32 "0xifg0fbh2rj3jn5i504c0qh51dlzgr1l2k962nh38synjhq9csc"))))
     (build-system pyproject-build-system)
     (arguments
      (list
-      #:phases
-      #~(modify-phases %standard-phases
-          (replace 'check
-            (lambda* (#:key tests? test-flags #:allow-other-keys)
-              (when tests?
-                (invoke "python" "setup.py" "build_ext" "--inplace")
-                ;; Step out of the source directory to avoid interference; we
-                ;; want to run the installed code with extensions etc.
-                (with-directory-excursion "/tmp"
-                  (apply invoke "pytest" "-v" test-flags))))))))
+      #:test-flags #~(list "--pyargs" "synphot")))
+    (native-inputs
+     (list python-pytest
+           python-pytest-astropy
+           python-setuptools
+           python-setuptools-scm
+           python-wheel))
     (propagated-inputs
      (list python-astropy
            python-dust-extinction
            python-numpy
            python-scipy
            python-specutils ))
-    (native-inputs
-     (list python-pytest
-           python-pytest-astropy
-           python-setuptools-scm))
     (home-page "https://github.com/spacetelescope/synphot_refactor")
     (synopsis "Synthetic photometry using Astropy")
     (description
-- 
2.46.0

